src/main.rs

```diff
@@ -1,5 +1,11 @@
 #![windows_subsystem = "windows"]
 
+#![allow(clippy::new_without_default)]
+#![allow(clippy::type_complexity)]
+#![allow(clippy::too_many_arguments)]
+#![allow(clippy::comparison_chain)]
+#![allow(clippy::await_holding_refcell_ref)]
```

Comment:
those probably, require refactors. `new_without_default` could be fixed by a few `derive(Default)` if desired.

Reply:
This lgtm, thanks!